src/apps/search/panels/BasePanel.tsx

```diff
@@ -1,77 +1,312 @@
-  const { data: { media_contents: mediaContents = [] } = {} } = useLoader(onLoadMedia);
+  const onLoadPeople = useCallback(() => Service.fetchRelatedPeople(id, { per_page: 0}), [id]);
+  const { data: { people = [] } = {} } = useLoader(onLoadPeople, null, [id]);
 
   /**
    * Loads the related place records from the Core Data API.
    */
-  const onLoadPlaces = useCallback(() => Service.fetchRelatedPlaces(id), [id]);
-  const { data: { places = [] } = {} } = useLoader(onLoadPlaces);
+  const onLoadPlaces = useCallback(() => Service.fetchRelatedPlaces(id, { per_page: 0 }), [id]);
+  const { data: { places = [] } = {} } = useLoader(onLoadPlaces, null, [id]);
+
+  /**
+   * Loads the related taxonomies from the Core Data API.
+   */
+  const onLoadTaxonomies = useCallback(() => Service.fetchRelatedTaxonomies(id, { per_page: 0 }), [id]);
+  const { data: { taxonomies = [] } = {} } = useLoader(onLoadTaxonomies, null, [id]);
+
+  /**
+   * Loads the related works from the Core Data API.
+   */
+  const onLoadWorks = useCallback(() => Service.fetchRelatedWorks(id, { per_page: 0 }), [id]);
+  const { data: { works = [] } = {} } = useLoader(onLoadWorks, null, [id]);
+
+  /**
+   * Memo-izes the bounding box options.
+   */
+  const bboxOptions = useMemo(() => getBoundingBoxOptions(config.map.max_zoom), [config]);
+
+  /**
+   * Memo-izes the base record.
+   */
+  const item = useMemo(() => data && data[props.name], [data, props.name]);
 
   /**
    * Memo-izes the geometry.
    */
-  const geometry = useMemo(() => !_.isEmpty(places) && CoreDataUtils.toFeatureCollection(places), [places]);
+  const geometry = useMemo(() => {
+    if (props.resolveGeometry && item) {
+      return props.resolveGeometry(item);
+    }
+
+    return !_.isEmpty(places) && CoreDataUtils.toFeatureCollection(places);
+  }, [item, places, props.resolveGeometry]);
 
   /**
-   * Memo-izes the image.
+   * Memo-izes the related media items.
    */
-  const image = useMemo(() => getPreviewImage(mediaContents), [mediaContents]);
+  const manifest = useMemo(() => {
+    if (_.isEmpty(collection.items)) {
+      return [];
+    }
+
+    const count = _.reduce(collection.items, (memo, item) => memo + item.item_count, 0);
+    const title = t('relatedMedia', { count });
+
+    return [{
+      items: _.map(collection.items, (item) => item),
+      renderItem: (item) => (
+        <ManifestThumbnail
+          className='py-2 px-6'
+          itemCount={item.item_count}
+          name={_.first(item.label?.en)}
+          onClick={() => setManifestUrl(item.id)}
+          thumbnail={item.thumbnail}
+        />
+      ),
+      renderTitle: () => title,
+      title
+    }];
+  }, [collection]);
 
   /**
    * Memo-izes the name.
    */
   const name = useMemo(() => (item && props.renderName && props.renderName(item)) || item?.name, [item]);
 
+  /**
+   * Transforms all the related records.
+   */
+  const relatedRecords = useMemo(() => [
```

Comment:
How often will this value be recomputed? There's a lot going on here and it looks like, e.g. when `people` changes, we'll be re-running tons of computations on other models that won't be any different than last time. That's probably fine if this is only being run once in practice, but otherwise I wonder if there would be a way to memoize individual object properties instead of the whole object?

Reply:
👍 yeah, I think that's probably a better approach.